src/flb_pack.c

```diff
@@ -810,7 +810,7 @@ flb_sds_t flb_msgpack_raw_to_json_sds(const void *in_buf, size_t in_size)
             tmp_buf = flb_sds_increase(out_buf, realloc_size);
             if (tmp_buf) {
                 out_buf = tmp_buf;
-                out_size *= realloc_size;
+                out_size += realloc_size;
```

Comment:
good catch. what about to setting `out_size` with `flb_sds_alloc()` ?

https://github.com/fluent/fluent-bit/blob/master/include/fluent-bit/flb_sds.h#L67-L70

Reply:
Done in [087ccda](https://github.com/fluent/fluent-bit/pull/9193/commits/087ccda70f195852319f46a11951e7af64216741). Tested and it seems to work as expected.